src/widgets/pile.rs

```diff
@@ -164,6 +169,8 @@ pub struct PendingPiledWidget(Option<PiledWidget>);
 
 impl PendingPiledWidget {
     /// Place `widget` in the pile and returns a handle to the placed widget.
+    ///
+    /// See [`PiledWidget`] for important lifetime information.

```

Comment:
```suggestion
    /// When the last clone of the returned [`PiledWidget`] is dropped, `widget`
    /// will be removed from the pile. If it is the currently visible widget,
    /// the next widget in the pile will be made visible.
```

I prefer not making the user jump through links to read information like this. I've pasted the summary from `push` as the suggestion here.